register_users.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#!/usr/bin/python
+
+import urllib2, random, requests, urllib2, socket, socks, sys
+from bs4 import BeautifulSoup
+from time import sleep
+from lxml import html
+from faker import Faker
+fake = Faker()
+
+url = 'https://github.com/join'
+
+
+def create_connection(address, timeout=None, source_address=None):
+    sock = socks.socksocket()
+    sock.connect(address)
+    return sock
+
+
+def verifyTor():
+    global pure_ip
+    ipcheck_url = 'http://checkip.amazonaws.com/'
+    pure_ip = requests.get(ipcheck_url).text.replace('\n','')
+    socks.setdefaultproxy(socks.PROXY_TYPE_SOCKS5, '127.0.0.1', 9050)
+    socket.socket = socks.socksocket
+    socket.create_connection = create_connection
+
+    tor_ip = urllib2.urlopen(ipcheck_url).read().replace('\n','') # Tor IP
+    if pure_ip == tor_ip:
+        print '[-] Unsuccessful Tor Connection'
+       	sys.exit(1)
+    else:
+        print '[+] Tor Running Normally'
+
+
+def createAccount():
+	session_requests = requests.session()
+	result = session_requests.get(url)
+	tree = html.fromstring(result.text)
+	
+	authenticity_token = list(set(tree.xpath("//input[@name='authenticity_token']/@value")))[0]
+	timestamp = list(set(tree.xpath("//input[@name='timestamp']/@value")))[0]
+	timestamp_secret = list(set(tree.xpath("//input[@name='timestamp_secret']/@value")))[0]
+	
+	field = list(set(tree.xpath("//input[@class='form-control' and not(text())]/@name")))
+	for name in field:
+		if name.startswith('required_field'):
+			required_field = name
+	
+	email = fake.email().encode('utf-8')
+	password = fake.password(special_chars=False, digits=True, upper_case=True, lower_case=True).encode('utf-8')
+	identity = {
+		'email': email,
+		'username': email.split('@')[0],
+		'password': password
+	}
+	print identity
+
+	payload = {
+		'utf8': '%E2%9C%93',
+		'authenticity_token': authenticity_token,
+		'user[login]': identity['username'],
+		'user[email]': identity['email'],
+		'user[password]': identity['password'],
+		'source': 'form-join',
+		required_field: '',
+		'timestamp': timestamp,
+		'timestamp_secret': timestamp_secret
+	}
+	#print payload
+	
+	headers = {'User-Agent': 'Mozilla/5.0', 'referer': url}
+	
+	#r = requests.get(url)
+	cookies = result.cookies.get_dict()
+	
+	result = session_requests.post(
+		url = url, 
+		data = payload, 
+		headers = headers,
+		cookies = cookies
+	)
+	
+	#print result.headers
+	#print result.text.encode('utf-8')
+	#print ",".join(identity.values())
+
+	if 'already taken' not in result.text.encode('utf-8'):
+		with open('data/users.txt', 'a') as mf:
+			mf.write(",".join(identity.values()) + '\n')
```

Comment:
If you just want to export the identifies, collect them into an array and marshall them using some standard (e.g. the Python pickle or, preferably, json).